### PR TITLE
Make relative sdk paths work from anywhere

### PR DESCRIPTION
**Pull request by Giuspepe**
Fixes https://github.com/vfdigitalincubator/noa-frontend/pull/1272/files#r1003189566 (fyi @nohli)

### CODE REVIEW

File: sidekick_core/lib/sidekick_core.dart [hunk 1]
```diff
@@ -31,6 +31,8 @@ export 'package:sidekick_core/src/git.dart';
 export 'package:sidekick_core/src/repository.dart';
 export 'package:sidekick_core/src/sidekick_package.dart';
 
+part 'src/resolve_sdk_path.dart';
```
Comment (passsy) [inline sidekick_core/lib/sidekick_core.dart:34]: Don't use part imports, they should only be used by code generators

File: sidekick_core/lib/src/resolve_sdk_path.dart [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      ? dir
+      // resolve relative path relative from project root
+      : Repository
+          // /Users/foo/project-x/packages/custom_sidekick
+          .requiredCliPackage
+          // /Users/foo/project-x/packages
+          .parent
+          // /Users/foo/project-x
+          .parent
+          .directory(sdkPath);
```
Comment (passsy) [inline sidekick_core/lib/src/resolve_sdk_path.dart:29]: I don't think we can assume that the project root is always two levels up

File: sidekick_core/lib/sidekick_core.dart [hunk 3]
```diff
@@ -68,8 +73,8 @@ SidekickCommandRunner initializeSidekick({
     repository: repo,
     mainProject: mainProject,
     workingDirectory: Directory.current,
-    flutterSdk: flutterSdkPath == null ? null : Directory(flutterSdkPath),
-    dartSdk: dartSdkPath == null ? null : Directory(dartSdkPath),
+    flutterSdk: _resolveSdkPath(flutterSdkPath),
+    dartSdk: _resolveSdkPath(dartSdkPath),
```
Comment (passsy) [inline sidekick_core/lib/sidekick_core.dart:77]: You could use the same method as for the `mainProject`:

```dart
  mainProject =
        DartPackage.fromDirectory(repo.root.directory(mainProjectPath));
```

```dart
flutterSdk = Directory(repo.root.directory(flutterSdkPath));
```

Plus, add the `isAbsolute` check

**Comment by nohli** [comment]
LGTM

**Pull request merged**